### PR TITLE
OADP-1002: Ignore not found err for temp vsclass in cases of data mover without PVC

### PR DESCRIPTION
**Pull request by eemcmullan**

### CODE REVIEW

File: pkg/datamover/datamover.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ import (
 	"github.com/pkg/errors"
 	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
 	"golang.org/x/sync/errgroup"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
```
Comment (shubham-pampattiwar) [inline pkg/datamover/datamover.go]: NIT: rename the import alias as `apierrors` , keeps things consistent with the code base

File: pkg/datamover/datamover.go [hunk 2]
```diff
@@ -137,6 +138,12 @@ func DeleteTempVSClass(backupName string, tempVS snapshotv1listers.VolumeSnapsho
 	tempVSClassName := fmt.Sprintf("%s-snapclass", backupName)
 	tempVSClass, err := tempVS.Get(tempVSClassName)
 	if err != nil {
+		// ignore is not found err as it is possible to create data mover
```
Comment (shubham-pampattiwar) [inline pkg/datamover/datamover.go:141]: +1 on the comment

File: pkg/datamover/datamover.go [hunk 3]
```diff
@@ -137,6 +138,12 @@ func DeleteTempVSClass(backupName string, tempVS snapshotv1listers.VolumeSnapsho
 	tempVSClassName := fmt.Sprintf("%s-snapclass", backupName)
```
Comment (shubham-pampattiwar) [inline pkg/datamover/datamover.go:138]: Maybe add a little comment about what the tempVSClass ? Not a dealbreaker.

Comment (eemcmullan) [reply pkg/datamover/datamover.go:138]: There's a small comment about temp vsclass in the backup controller

File: pkg/datamover/datamover.go [hunk 4]
```diff
@@ -137,6 +138,12 @@ func DeleteTempVSClass(backupName string, tempVS snapshotv1listers.VolumeSnapsho
...
 	tempVSClass, err := tempVS.Get(tempVSClassName)
 	if err != nil {
+		// ignore is not found err as it is possible to create data mover
+		// backup without an existing PVC
+		// in which case this VSClass will not exist
+		if apierrors.IsNotFound(err) {
+			return nil
```
Comment (weshayutin) [inline pkg/datamover/datamover.go:145]: seems fine to me IF the only errors we need to be concerned about are apierrors.

**Review by shubham-pampattiwar — APPROVED** [review]

File: pkg/datamover/datamover.go (hunk 4 above)
Comment (eemcmullan) [reply pkg/datamover/datamover.go:145]: AFAIK for this particular issue this should be the only error we're concerned about

**Pull request merged**